PseudoGantry/src/main.cpp

```diff
@@ -0,0 +1,143 @@
+#define AXIS_LENGTH_COUNTS_X 2000
+#define AXIS_LENGTH_COUNTS_Y 2000
+
+//
+#define MOTOR_START_POSITION_X 0
+#define MOTOR_START_POSITION_Y 0
+
+PseudoEncoder pseudo_encoder_x;
+PseudoEncoder pseudo_encoder_y;
+PseudoAxis pseudo_axis_x;
+PseudoAxis pseudo_axis_y;
+
+void isr_motor_pulse_x()
+{
+    static uint32_t last_interrupt_time = 0;
+    uint32_t interrupt_time = micros();
+    if (interrupt_time - last_interrupt_time > 100) {
+          isr_motor_pulse(&pseudo_axis_x);
+    }
+    last_interrupt_time = interrupt_time;
+}
+
+void isr_motor_pulse_y()
+{
+    static uint32_t last_interrupt_time = 0;
+    uint32_t interrupt_time = micros();
+    if (interrupt_time - last_interrupt_time > 100) {
+          isr_motor_pulse(&pseudo_axis_y);
+    }
+    last_interrupt_time = interrupt_time;
+}
+
+void setup()
+{
+  pseudo_encoder_x = {
+      .motor_pulse_pin = MOTOR_AXIS_X,
+      .channel_a_out = ENCODER_OUT_X
+      };
+  // pseudo_encoder_y = {
+  //     .motor_pulse_pin = MOTOR_AXIS_Y,
+  //     .channel_a_out = ENCODER_OUT_Y
+  //     };
+
+  set_up_encoder(&pseudo_encoder_x, &isr_motor_pulse_x);
+  // set_up_encoder(&pseudo_encoder_y, &isr_motor_pulse_y);
+
+  pseudo_axis_x = {
+      .axis_name = "X",
+      .encoder = pseudo_encoder_x,
+      .axis_length_counts = AXIS_LENGTH_COUNTS_X,
+      .motor_position_current = MOTOR_START_POSITION_X,
+      .motor_position_default = MOTOR_START_POSITION_X,
+      .motor_dir_pin = MOTOR_DIR_X,
+      .skip_counter = 0,
+      .steps_for_ratio = STEPS,
+      .counts_for_ratio = COUNTS,
+      .changes_to_skip = 2 * (STEPS - COUNTS),
+      .ls_home = {
+        .output_pin = LIMIT_SW_OUT_HOME_X,
+        .status = UNRESSED
+      },
+      .ls_far = {
+        .output_pin = LIMIT_SW_OUT_FAR_X,
+        .status = UNRESSED
+      }
+      };
+
+  // pseudo_axis_y = {
+  //     .axis_name = "Y",
+  //     .encoder = pseudo_encoder_y,
+  //     .axis_length_counts = AXIS_LENGTH_COUNTS_Y,
+  //     .motor_position_current = MOTOR_START_POSITION_Y,
+  //     .motor_position_default = MOTOR_START_POSITION_Y,
+  //     .motor_dir_pin = MOTOR_DIR_Y,
+  //     .skip_counter = 0,
+  //     .steps_for_ratio = STEPS,
+  //     .counts_for_ratio = COUNTS,
+  //     .changes_to_skip = 2 * (STEPS - COUNTS),
+  //     .ls_home = {
+  //       .output_pin = LIMIT_SW_OUT_HOME_Y,
+  //       .status = UNRESSED
+  //     },
+  //     .ls_far = {
+  //       .output_pin = LIMIT_SW_OUT_FAR_Y,
+  //       .status = UNRESSED
+  //     }
+  //     };
+
+  delay(1000);
+
+  reset_pseudo_axis(&pseudo_axis_x);
+  // reset_pseudo_axis(&pseudo_axis_y);
+
+  // setup serial for monitoring
+  Serial.begin(BAUDRATE);
+
+  // pretend to be a motor
+  pinMode(MOTOR_DIR_X, INPUT_PULLUP);
+  pinMode(PWM_PIN, OUTPUT);
+}
+
+void loop()
+{
+  // put your main code here, to run repeatedly:
+  dump_data(&pseudo_axis_x);
+  delay(5000);
+  analogWrite(PWM_PIN, 122);
+  delay(delay_millis);
+  analogWrite(PWM_PIN, 0);
```

Comment:
I assume this was for your "acting as a motor" testing?
Can probably replace this with just:
```c
dump_data(&pseudo_axis_x);
dump_data(&pseudo_axis_y);
delay(1000);
```
or something similar